plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/CDKConstantUtility.java

```diff
@@ -10,29 +10,49 @@
  ******************************************************************************/ 
 package org.jboss.tools.openshift.cdk.server.core.internal;
 
-import org.eclipse.debug.core.ILaunchConfiguration;
+import java.io.File;
+
 import org.eclipse.core.runtime.CoreException;
 import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.debug.core.ILaunchConfiguration;
 import org.eclipse.wst.server.core.IServer;
 import org.jboss.tools.openshift.cdk.server.core.internal.adapter.controllers.IExternalLaunchConstants;
+import org.jboss.tools.openshift.internal.common.core.util.CommandLocationProvider;
 
-// TODO - allow customization of this location
 public class CDKConstantUtility {
 	private static final String VAGRANT_LOCATION_LINUX = "/usr/bin/vagrant";
-
+	
+	// Seems weird?  See https://github.com/mitchellh/vagrant/issues/1652
+	private static final String VAGRANT_LOCATION_WINDOWS = "C:\\HashiCorp\\Vagrant\\bin\\vagrant.exe";
+	
+	private static CommandLocationProvider locationProvider;
+	
 	public static String getVagrantLocation() {
-		return VAGRANT_LOCATION_LINUX;
+		return findVagrantLocation();
 	}
 	
 	public static String getVagrantLocation(IServer server) {
-		try {
-			ILaunchConfiguration lc = server.getLaunchConfiguration(false, new NullProgressMonitor());
-			if( lc != null ) {
-				return lc.getAttribute(IExternalLaunchConstants.ATTR_LOCATION, getVagrantLocation());
+		if( server != null ) {
+			try {
+				ILaunchConfiguration lc = server.getLaunchConfiguration(false, new NullProgressMonitor());
+				if( lc != null ) {
+					String ret = lc.getAttribute(IExternalLaunchConstants.ATTR_LOCATION, (String)null);
+					if( ret != null )
+						if( new File(ret).exists())
+							return ret;
+				}
+			} catch(CoreException ce) {
+				// ignore, this is non-critical
 			}
-		} catch(CoreException ce) {
-			// ignore, this is non-critical
 		}
-		return VAGRANT_LOCATION_LINUX;
+		return findVagrantLocation();
+	}
+	
+	private static String findVagrantLocation() {
+		if( locationProvider == null ) {
+			locationProvider = new CommandLocationProvider("vagrant.exe", "vagrant", 
+					VAGRANT_LOCATION_WINDOWS, VAGRANT_LOCATION_LINUX);
+		}
+		return locationProvider.getLocation(10 * 1000);
```

Comment:
I know, but which/where should return *fast* anyway, so if it takes more than 1 sec, then it's doomed anyway